models/base/meta_document.py

```diff
@@ -0,0 +1,48 @@
+import sys
+
+from models.base.field import Field
+
+
+class MetaDocument(type):
+    """
+    A metaclass for documents to register all fields and identify the primary key.
+    """
+
+    class MultiplePrimaryKeyError(Exception):
+        def __init__(self):
+            super().__init__(f"Only one primary key is allowed")
```

Comment:
What is 'f' referencing/returning here?

Reply:
Good catch, it's useless. 
(I think you meant to comment on #27 though, i will update there)